packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -198,6 +202,15 @@ export class AppManager implements BackstageApp {
     return this.components;
   }
 
+  getExtensions(): ComponentExtensions {
+    return [
+      // Plugin-provided extensions
+      ...this.getPlugins().flatMap(plugin => plugin.extensions ?? []),
```

Comment:
The ordering of the plugins list is deterministic, but not really from the users point of view. i.e. small changes to how the app is wired can have a large impact on the ordering of plugins. I think it's important that if we were to do anything like this, the ordering of the plugins has no affect on the rendered extensions.

Reply:
Both you and @kuangp prefer opt-in, and if we go that way, I think this won't be an issue. But you're right, if we keep this opt-out, it'll be quite confusing how multiple extension of the same component end up with a different result because of plugin loading order changes.